mozilla/foundation.mozilla.org
# Refactor search.js

Closes #7988 but only as the refactor (with tests) part. Actual code cleaning (consolidation, optmization, etc) will not be done as part of this issue/PR

### Testing:

- check out this branch
- edit your `.env` to include `RANDOM_SEED=530910203` to make sure the fake data you're going to generate is the one the integration test expects
- run `inv new-db` (to make sure the fake data is up to date, as this PR updates what that looks like) 
- run `docker-compose up` in one terminal
- run `npm run tailwind` in another terminal once docker's up and running
- run `npm run playwright` in a third terminal, once the server and tailwind compile are both up and running in watch mode.

This should complete two integration tests - one for the fomo homepage, and one for PNI search. Both should pass.

- open `source/js/buyers-guide/search.js` in your code editor
- open https://raw.githubusercontent.com/mozilla/foundation.mozilla.org/main/source/js/buyers-guide/search.js and copy its content into search.js in your editor, overwriting the file contents
- save
- run `npm run playwright` again

This will restore search.js to its original (i.e. current live) form, but because this was a pure refactor _everything should work the same_ and so running `npm run playwright` again should still pass on the original code.
